### ISSUE TITLE
Data viewer gives subscript-out-of-bounds when searching

### ISSUE THREAD

**Issue by aronatkins** [rstudio/rstudio#1833]
RStudio version 1.2.194

```r
> j <- jsonlite::fromJSON('[{"first":"bob","last":"smith"}]')
> View(j)
```

In the viewer, search for "ann". You will see errors in the console like:
```
Error in info[[1]] : subscript out of bounds
Error in info[[1]] : subscript out of bounds
```

The UI correctly shows that there are no results.
<img width="674" alt="screen shot 2017-12-09 at 11 10 01 am" src="https://user-images.githubusercontent.com/362187/33797182-868c2300-dcd1-11e7-98c1-d890e60b39df.png">

**Comment by ronblum** [comment]
This works correctly in 1.1.393, and reproducible in 1.2.194.

**Comment by jmcphers** [comment]
Fixed in #2230.

**Comment by dfalty** [comment]
Verified fixed 1.2.359.
